src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -81,6 +81,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Message message = new Message(user, text);
     datastore.storeMessage(message);
 
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+        
+    message = new Message(user, textWithImagesReplaced);
+    datastore.storeMessage(message);
+
```

Comment:
Looks good to me but this was also handled on the front end by me. Did you make sure to remove the front end code and make sure it still works on the feed and username pages?